src_c/geometry.c

```diff
@@ -8,6 +8,106 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    PyObject **farr;
+    Py_ssize_t farr_length;
+    Py_ssize_t loop;
+    double max_t;
+    pgLineBase line;
+
+    if (nargs == 4) {
+        if (!pg_TwoDoublesFromObj(args[0], &line.x1, &line.y1)) {
+            return RAISE(PyExc_TypeError,
+                         "the origin requires a pair of floats");
```

Comment:
The error message is too generic and a bit misleading... I'd put something like: `"Invalid ray origin value, must be a pair of numeric values"`